lib/appmap/service/guesser.rb

```diff
@@ -6,15 +6,8 @@ class Guesser
       POSSIBLE_PATHS = %w[app/controllers app/models lib]
       class << self
         def guess_name
-          reponame = lambda do
-            next unless File.directory?('.git')
-
-            repo_name = `git config --get remote.origin.url`.strip
-            repo_name.split('/').last.split('.').first unless repo_name == ''
-          end
-          dirname = -> { Dir.pwd.split('/').last }
-
-          reponame.() || dirname.()
+          return `basename $(git rev-parse --show-toplevel)`.strip if File.directory?('.git')
```

Comment:
We should use a Ruby function (`Pathname`?) rather than `basename` because basename may not be available on all platforms (e.g. Windows).

Reply:
Yeah. My suggestion was meant to show what we wanted to do, not how it should be implemented.

Reply:
Good catch! Fixed.